db/migrate/20210302165104_create_bookings.rb

```diff
@@ -0,0 +1,15 @@
+class CreateBookings < ActiveRecord::Migration[6.0]
+  def change
+    create_table :bookings do |t|
+      t.references :user, null: false, foreign_key: true
+      t.datetime :start_time
+      t.datetime :end_time
+      t.references :lesson, null: false, foreign_key: true
+      t.references :review, null: false, foreign_key: true
+      t.references :payment, null: false, foreign_key: true
+      t.references :report, null: false, foreign_key: true
```

Comment:
The issue you might have faced would be here. Booking belongs to a lesson, but only optionally has a review, a payment or a report. So `null: false` wasn't applicable.

Reply:
@DeFiDegen